### PR TITLE
OCSADV-518: Compute set of systems used by a sequence.

### PR DESCRIPTION
**Pull request by jluhrs**
This is the first step in implementing the resource management: finding out what systems will be used to run a specific sequence.

### CODE REVIEW

**Comment by cquiroz** [comment]
Looks good, you may want to have a description for your PR though :+1:

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ExecutorImpl.scala [hunk 1]
```diff
@@ -24,15 +19,22 @@ class ExecutorImpl private (cancelRef: TaskRef[Set[SPObservationID]], stateRef:
   private def go(id: SPObservationID): Task[Boolean] = 
     cancelRef.get.map(!_(id))
 
-  private def initialExecState(sequenceConfig: ConfigSequence): ExecState =
-    ExecState.initial(sequenceConfig.getAllSteps.toList.map(Step.step))
+  def sequence(sequenceConfig: ConfigSequence): (Set[System], List[Step.Step]) =
+    sequenceConfig.getAllSteps.toList.map(Step.step).unzip match {
+      case (s, l) => (Set(s: _*).flatMap(identity), l)
+    }
```
Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ExecutorImpl.scala]: `Set(s: _*).flatMap(identity)` → `s.suml`

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala [hunk 2]
```diff
@@ -42,10 +42,10 @@ object Step {
 
     instrument map { a => {
         val systems = List(Tcs(TcsControllerEpics), a)
-  //      val systems = List(TCS(TcsControllerSim), a)
-        step(systems.map(_.configure(config)), a.observe(config))
+  //      val systems = List(Tcs(TcsControllerSim), a)
+        (Set(systems: _*), step(systems.map(_.configure(config)), a.observe(config)))
```
Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala]: `Set(systems: _*)` → `systems.toSet`

**Comment by tpolecat** [comment]
Looks good modulo a couple simplifying identities.

**Pull request merged**